### ISSUE TITLE
iOS picker renders incorrectly in landscape

### ISSUE THREAD

**Issue by steemcb-phorest** [henninghall/react-native-date-picker#478]
**Describe the bug**
Setup a modal picker and run your application in landscape mode. You will see that the picker renders off-centre and only the left-most field is touchable.

![Simulator Screen Shot - iPhone 13 Pro - 2022-03-14 at 10 53 10](https://user-images.githubusercontent.com/70574674/158158123-190b624f-dded-4391-a31d-af7853d48445.png)

**Expected behavior**
Picker should render centred and all fields should be accessible

**To Reproduce**

```javascript
export default class App extends Component {

  state = { date: new Date() }

  render = () =>
    <DatePicker
      date={this.state.date}
      onDateChange={date => this.setState({ date })}
      modal={true}
    />

}
```

**Smartphone (please complete the following information):**
 - OS: iOS iPhone 15.0 , iPad not affected
 - React Native version 0.65.1
 - react-native-date-picker version 4.2.0

**Comment by 8ur4k** [comment]
hey I have the same bug, do you have any workarounds for this?

**Comment by scipilot** [comment]
I also encountered this bug, in React 17.0, RN 0.67.4, RNDP 4.2.1, 
on iPhone 8 simulator, iOS 13.2
on iPhone 11 simulator, iOS 13.2
on iPhone 13 simulator, iOS 15.5
on iPhone SE real, iOS 15.4.1

I also find it does not happen on iPads (Mini real, Pro sim)

Interestingly, if you open the picker in portrait orientation, then turn the phone to landscape it's rendered correctly when it flips, until you close and re-open it.

**Comment by scipilot** [comment]
This seems to be a bug in the underlying RNDatePickerManager not this wrapping library. 

The simplest code I've found reproduces it is: 

```
    NativeModules.RNDatePickerManager.openPicker(
        {
            minimumDate:Date.now(),
            maximumDate:Date.now(),
            mode:"date",
            modal:true,
            open:true,
            date:Date.now(),
            title:"The title",
            confirmText:"OK",
            cancelText:"CANCEL",
            theme:"light",
        },
        ()=>{},
        ()=>{}
    )
```

**Comment by chan3111** [comment]
I am also experiencing this, it looks like the picker is setting a width of the screen width - 15 pushing it off to the right.

It's somewhere around this code
https://github.com/henninghall/react-native-date-picker/blob/7e9a5beb09174811697a7925c96dd209923fc4a7/ios/RNDatePicker/RNDatePickerManager.m#L82

I attempted a fix but my skills in Objective-C are not nearly good enough. For now going to just implement my own modal and render it inline.

**Comment by henninghall** [comment]
Fixed in v4.2.3
